File: Animal Shelter Opdracht/C# versie/animal_shelter-abstract_interfaces-649f03d658d7546000a2cadc0e118ada4f610607/Animal Shelter/Animals/Cat.cs
using System;
using System.Diagnostics.Contracts;

namespace Animal_Shelter.Animals
{
    public class Cat : Animal
    {
        public string BadHabits { get; private set; }

        public Cat(string name, Gender gender, string badHabits)
            : base(name, gender)
        {
            this.BadHabits = badHabits;
        }

        public override string ToString()
        {
            return base.ToString() +
                $", bad habits: {this.BadHabits.ToLower()}";
        }
    }
}

File: src/Animals/Cat.java
package Animals;

import java.util.ArrayList;
import java.util.List;

public class Cat extends Animal
{

    public String BadHabits;
    private int maxPrice = 350;

    public Cat(String name, Gender gender, String badHabits)
    {
        super(name, gender, 350);

        BadHabits = badHabits;
        int discount = badHabits.length() * 20;
        int price = maxPrice - discount;
        if( price < 35)
        {
            this.price = 35;
        }
        else
        {
            this.price = price;
        }

    }

    public String getBadHabits()
    {
        return BadHabits;
    }

    public void setBadHabits(String badHabits)
    {
        BadHabits = badHabits;
    }

}
